lib/AST/Expr.cpp

```diff
@@ -2224,24 +2224,29 @@ TypeExpr *TypeExpr::createForMemberDecl(DeclNameLoc ParentNameLoc,
   return new (C) TypeExpr(TR);
 }
 
-TypeExpr *TypeExpr::createForMemberDecl(DeclRefTypeRepr *ParentTR,
-                                        DeclNameLoc NameLoc, TypeDecl *Decl) {
+TypeExpr *TypeExpr::createForMemberDecl(TypeRepr *ParentTR, DeclNameLoc NameLoc,
+                                        TypeDecl *Decl) {
   ASTContext &C = Decl->getASTContext();
 
-  // Create a new list of components.
-  SmallVector<IdentTypeRepr *, 2> Components;
-  if (auto *MemberTR = dyn_cast<MemberTypeRepr>(ParentTR)) {
-    auto MemberComps = MemberTR->getMemberComponents();
-    Components.append(MemberComps.begin(), MemberComps.end());
-  }
-
   // Add a new component for the member we just found.
   auto *NewComp = new (C) SimpleIdentTypeRepr(NameLoc, Decl->createNameRef());
   NewComp->setValue(Decl, nullptr);
-  Components.push_back(NewComp);
 
-  auto *TR =
-      MemberTypeRepr::create(C, ParentTR->getBaseComponent(), Components);
+  TypeRepr *TR = nullptr;
+  if (auto *DeclRefTR = dyn_cast<DeclRefTypeRepr>(ParentTR)) {
+    // Create a new list of components.
+    SmallVector<IdentTypeRepr *, 2> Components;
```

Comment:
FWIW `SmallVector` doesn’t require a size anymore, so this could be `SmallVector<IdentTypeRepr *>`. IIUC unless you have very good reasons to chose an explicit parameter here, I think the default should be good enough. I don’t have super strong opinions though.

Reply:
The default here would be 6, which I think is a bit too much. A nested type of depth 3 in expression context is already  feels like edge-case land. But perhaps 4 is more reasonable choice than 2. Thoughts?

Reply:
I don’t really have strong opinions and I don’t think it makes a huge difference either way. I just got into the habit of omitting the size whenever I don’t really know. 

But really, I’m happy with either value here. I just wasn’t sure if you knew that this parameter can be omitted now.

Reply:
> I just wasn’t sure if you knew that this parameter can be omitted now

It’s nice to be reminded 🙂. I do recall spotting this whilst reading through the interface quite some time ago now that I think about it.

> But really, I’m happy with either value here.

Let’s go with 4 then.